### ISSUE TITLE
Internal error in simple self-join query

### ISSUE THREAD

**Issue by hlinnaka** [greenplum-db/gpdb#2100]
```
postgres=# create table parttab (id int4) partition by range (id) (start (1) end (100) every (20));
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 'id' as the Greenplum Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
NOTICE:  CREATE TABLE will create partition "parttab_1_prt_1" for table "parttab"
NOTICE:  CREATE TABLE will create partition "parttab_1_prt_2" for table "parttab"
NOTICE:  CREATE TABLE will create partition "parttab_1_prt_3" for table "parttab"
NOTICE:  CREATE TABLE will create partition "parttab_1_prt_4" for table "parttab"
NOTICE:  CREATE TABLE will create partition "parttab_1_prt_5" for table "parttab"
CREATE TABLE
postgres=# select * from parttab a, parttab b where a.id = b.id;
ERROR:  Unexpected internal error (execMain.c:5659)
```

**Comment by hlinnaka** [comment]
It's not restricted to a self-join, you get the same error if you join two different but identical tables.

**Comment by hlinnaka** [comment]
Fixed by commit e378d84bf32922b324a8ce6e8a66be358a0b805f
